server/src/main/java/org/elasticsearch/common/io/stream/RecyclerBytesStreamOutput.java

```diff
@@ -218,9 +230,7 @@ public BytesReference bytes() {
     }
 
     private void ensureCapacity(int bytesNeeded) {
-        if (bytesNeeded > pageSize - currentPageOffset) {
-            ensureCapacityFromPosition(position() + bytesNeeded);
-        }
+        ensureCapacityFromPosition(position() + bytesNeeded);
```

Comment:
Can we `assert bytesNeeded > pageSize - currentPageOffset` here?

Reply:
Done :)